### PR TITLE
Synbatch

### PR DESCRIPTION
**Pull request by Bonbon-Tang**
## Description
 新增，Synchro-BatchNorm-with-count 函数，根据物理意义以及cnnl内部算子输入要求，更改了with_counts函数的ci input（只有正整数才有意义，正小数会被floor操作，cnnl的负小数会报错，实际情况下应该避免）

## Checklist
**Before PR**:

- [x] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: impl/camb/functions/SynBatchNorm.cpp [hunk 1]
```diff
@@ -354,5 +354,82 @@ DIOPI_API diopiError_t diopiBatchNormStats(diopiContextHandle_t ctx, diopiTensor
...
+                                                          meanAllDesc.get(),
+                                                          meanAllTr.data(),
+                                                          invstdAllDesc.get(),
+                                                          invstdAllTr.data(),
+                                                          runningMeanDesc.get(),
+                                                          runningMeanTr.defined() ? runningMeanTr.data() : nullptr,
+                                                          runningVarDesc.get(),
+                                                          runningVarTr.defined() ? runningVarTr.data() : nullptr,
+                                                          momentum,
+                                                          eps,
```
Comment (PetrelYy) [inline impl/camb/functions/SynBatchNorm.cpp:418]: 和357 入参对齐方式不同，统一下？

**Review by PetrelYy — APPROVED** [review]
LGTM

File: impl/camb/functions/SynBatchNorm.cpp [hunk 2]
```diff
@@ -354,5 +354,82 @@ DIOPI_API diopiError_t diopiBatchNormStats(diopiContextHandle_t ctx, diopiTensor
     return diopiSuccess;
 }
 
+DIOPI_API diopiError_t diopiBatchNormGatherStatsWithCounts(diopiContextHandle_t ctx, diopiTensorHandle_t mean, diopiTensorHandle_t invstd,
+                                                           diopiConstTensorHandle_t input, diopiConstTensorHandle_t meanAll, diopiConstTensorHandle_t invstdAll,
+                                                           diopiTensorHandle_t runningMean, diopiTensorHandle_t runningVar, float momentum, float eps,
+                                                           diopiConstTensorHandle_t counts) {
```
Comment (yangbofun) [inline impl/camb/functions/SynBatchNorm.cpp:360]: 函数内对counts中存在小于0的元素，做一下check，与报错提示。

**Review by yangbofun — APPROVED** [review]

File: diopi_test/python/conformance/diopi_configs.py [hunk 3]
```diff
@@ -218,9 +218,9 @@
                 },
                 {
                     "ins": ["count_all"],
-                    "shape": ((16,), (448,), (288,), (64,)),
+                    "shape": ((2,), (7,), (3,), (4,)),
                     "requires_grad": [False],
-                    "gen_fn": Genfunc.randn,
+                    "gen_fn": dict(fn=Genfunc.randint, low=2, high=6),
```
Comment (yangbofun) [inline diopi_test/python/conformance/diopi_configs.py:223]: 这里先写randint，等diopi_test重构后，这里写生成正的浮点数。

Comment (Bonbon-Tang) [reply diopi_test/python/conformance/diopi_configs.py:223]: 好的

Comment (LeungChiNan) [inline diopi_test/python/conformance/diopi_configs.py:223]: nv下输入负数也会报错吗，如果不会则也需要覆盖到该测例

Comment (Bonbon-Tang) [reply diopi_test/python/conformance/diopi_configs.py:223]: nv输入和cnnl测过，目前结果完全不一致，负数理论上没有意义，所以剔除了

Comment (yangbofun) [reply diopi_test/python/conformance/diopi_configs.py:223]: nv下为负数得到的值是没有意义的。函数接口的定义里，这里也要说明，这个count不能为负数。

**Review by LeungChiNan — APPROVED** [review]
